### PR TITLE
Restructure pca matrix

### LINKED ISSUE #759: Allow different data types as input to PCA-scaling
Different data types will have different number of data points.

### LINKED ISSUE #796: Restructure input for PCA calculation in scaling job
Input to PCA-scaling should be restructured such that all notion of time and space is flattened, the input matrix should be structured as individual points on the x-axis and realizations on the y-axis.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #796 
Resolves #759

### CODE REVIEW

File: python/res/enkf/jobs/scaling/measured_data.py [hunk 1]
```diff
@@ -9,58 +9,79 @@
...
         Adds simulated and observed data and returns a dataframe where ensamble members will
-        have a data key, observed data will be prefixed with OBS_ and observated standard
-        deviation will be prefixed with STD_.
+        have a data key, observed data will be named OBS and observated standard
+        deviation will be named STD.
         """
         measured_data = pd.DataFrame()
         case_name = ert.getEnkfFsManager().getCurrentFileSystem().getCaseName()
-        for key in observation_keys:
+        for key, index_list in zip(observation_keys, index_lists):
```
Comment (jondequinor) [inline python/res/enkf/jobs/scaling/measured_data.py:48]: Maybe comment on why zip is used here?

File: python/res/enkf/jobs/scaling/measured_data.py [hunk 2]
```diff
@@ -202,30 +223,32 @@ def _combine_filters(filters):
...
-        return data.set_index([[observation_key] * len(data)])
+        return data.set_index(data.index.values)
 
     @staticmethod
-    def _add_summary_observations(ert, data_key, observation_key, case_name):
+    def _add_summary_observations(ert, data_key, case_name):
         data = SummaryObservationCollector.loadObservationData(
             ert, case_name, [data_key]
         ).transpose()
         data = data.set_index(
```
Comment (jondequinor) [inline python/res/enkf/jobs/scaling/measured_data.py:236]: I think an explanation would help here. Why are you doing this?

**Review by jondequinor — APPROVED** [review]
Could you elaborate on your approach here? I am having trouble connecting the linked issues to the changeset.

Besides the things I've asked for comments on, it looks OK!

File: python/res/enkf/jobs/scaling/measured_data.py (hunk 1 above)
Comment (oyvindeide) [reply python/res/enkf/jobs/scaling/measured_data.py:48]: The change introduced here will allow the user to input an index list for each key, which wasn't possible before as a common was given for all keys. The calling function should make sure that an index list (could be None) exists for each key.

File: python/res/enkf/jobs/scaling/measured_data.py (hunk 2 above)
Comment (oyvindeide) [reply python/res/enkf/jobs/scaling/measured_data.py:236]: Will add it to the code :+1:

File: python/res/enkf/jobs/scaling/measured_data.py (hunk 1 above)
Comment (oyvindeide) [reply python/res/enkf/jobs/scaling/measured_data.py:48]: :+1:

**Pull request merged**